core/dbt/events/base_types.py

```diff
@@ -88,6 +102,18 @@ def get_pid(self) -> int:
             self.pid = os.getpid()
         return self.pid
 
+    def get_node_info(self):
+        return None
+
+
+class NodeInfo(Event, metaclass=ABCMeta):
+    report_node_data: Any
```

Comment:
hmm lets see if we can make this type annotation tighter than `Any`. Looking at the call sites below, I think they're all subclasses of some parent Node type?